src/magmaviz/boxplot.py

```diff
@@ -20,5 +24,52 @@ def boxplot(df, x, y, facet=False):
     Examples
     --------
     >>> from magmaviz.magmaviz import boxplot
-    >>> boxplot(mtcars, horsepower, cars, facet=True)
+    >>> boxplot(cars, "Miles_per_Gallon", "Origin", facet=True)
     """
+
+    # checking that type of column name for x is a string
+    if not isinstance(x, str):
+        raise TypeError("'x' should be of type 'str'.")
+    # checking that type of column name for y is a string
+    if not isinstance(y, str):
+        raise TypeError("'y' should be of type 'str'.")
+    # checking that type for facet is a boolean
+    if not isinstance(facet, bool):
+        raise TypeError("'facet' should be of type 'boolean'.")
+    # checking that the dataframe is a pandas dataframe type
+    if not isinstance(df, pd.core.frame.DataFrame):
+        raise TypeError(
+            "'df' should be of type 'pandas.core.frame.DataFrame', a pandas dataframe."
+        )
+
+    # checking if x is a column name in the dataframe
+    assert x in list(
+        df.columns
+    ), "This column specified for 'x' does not exist in the dataframe."
+    # checking if y is a column name in the dataframe
+    assert y in list(
+        df.columns
+    ), "This column specified for 'y' does not exist in the dataframe."
+
```

Comment:
I think you should also add tests to check if column 'x' from the data frame is of type 'str', and column 'y' is of type numeric? Because a user may pass say numeric column names to both 'x' and 'y' and all of these tests would pass but it would still give an error?